api/configuration/v1alpha1/kong_tags_types.go

```diff
@@ -0,0 +1,6 @@
+package v1alpha1
+
+// Tags is a type for a list of tags applied to a Kong entity.
+// +kubebuilder:validation:MaxItems=20
```

Comment:
Kong gateway itself does not have the constraints of 20 tags and max length = 128. So the validation rules may not apply to non-Konnect environment.

Reply:
Being more restrictive we'd still satisfy the constraints of both environments. I can see the following options:

- we implement the more restrictive rules (in this case - Konnect) and stay always safe
- we implement the less restrictive rules (in this case - Kong) and we're safe with Kong, but we fail in runtime with Konnect (and propagate errors in status)
- we do not implement any constraints on the CRD level, rely on server-side validation rules, and propagate issues in entities' status

Reply:
I'd go with the first option as the explicit `Tags` field is added to CRDs now, so we do not break anything for users and we get a safety guarantee quite cheaply.

Reply:
I think it is OK to apply a more restrict validation rule as it is a "new" field. If we really have such requirement to apply looser constraint (for example, using the resources to operate Kong gateway), we could update the validation rules then.